### PR TITLE
#6306: Enable N150,N300 ttnn unit tests in CI Regressions; disable failing ones

### PR DESCRIPTION
**Pull request by cfjchu**

### CODE REVIEW

**Review by tt-rkim — APPROVED** [review]

**Review by eyonland — APPROVED** [review]

**Review by arakhmati — APPROVED** [review]

File: tests/ttnn/unit_tests/operations/test_conv2d.py [hunk 1]
```diff
@@ -747,6 +752,9 @@ def test_sd_conv(
         )
 
 
+@skip_for_wormhole_b0(
```
Comment (tt-nshanker) [inline tests/ttnn/unit_tests/operations/test_conv2d.py:755]: I do not expect SD conv unit tests to fail on WH. The SD model runs on WH and there are no l1 fitting issues there.

File: tests/ttnn/unit_tests/operations/test_conv2d.py [hunk 2]
```diff
@@ -988,6 +996,7 @@ def test_unet_conv(
     )
 
 
+@skip_for_wormhole_b0("Issue #6989: AttributeError: 'NoneType' object has no attribute 'enable_program_cache'")
```
Comment (tt-nshanker) [inline tests/ttnn/unit_tests/operations/test_conv2d.py:999]: Unet convs in UNet model runs on WH without errors. Not sure why unit test is failing. @nsmithtt we don't see an error on WH right?

File: tests/ttnn/unit_tests/operations/test_conv2d.py (hunk 1 above)
Comment (cfjchu) [reply tests/ttnn/unit_tests/operations/test_conv2d.py:755]: We synced offline.. @tt-nshanker mentioned some of these failing permutations are easy fixes and will follow-up on some of these so we can add them to the test-coverage

**Comment by cfjchu** [comment]
Clean post-commit run: https://github.com/tenstorrent-metal/tt-metal/actions/runs/8528783943/job/23363436964

**Review by tt-nshanker — APPROVED** [review]

**Pull request merged**